### PR TITLE
[->master] LoginController, KingdomController(GET, PUT), DTOs, Flyway, refactor(error handling)

### PR DESCRIPTION
**Pull request by akmajoros**

### CODE REVIEW

File: src/main/java/com/greenfox/tribesoflagopus/backend/controller/KingdomController.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+public class KingdomController {
+
+  private final KingdomService kingdomService;
+
+  @Autowired
+  public KingdomController(KingdomService kingdomService) {
+    this.kingdomService = kingdomService;
+  }
+
+  @GetMapping("{userId}/kingdom")
```
Comment (seed0forever) [inline src/main/java/com/greenfox/tribesoflagopus/backend/controller/KingdomController.java:26]: TODO: mapped path should begin with a `/`

File: src/main/java/com/greenfox/tribesoflagopus/backend/controller/KingdomController.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  public ResponseEntity<JsonDto> showKingdom(@PathVariable Long userId) {
+    return kingdomService.showKingdom(userId);
+  }
+
+  @PutMapping(value = "{userId}/kingdom",
```
Comment (seed0forever) [inline src/main/java/com/greenfox/tribesoflagopus/backend/controller/KingdomController.java:31]: TODO: mapped path should begin with a `\`

File: src/main/java/com/greenfox/tribesoflagopus/backend/BackendApplication.java [hunk 3]
```diff
@@ -23,27 +23,19 @@ public static void main(String[] args) {
...
-              .build();
+							.username("Noemi")
+							.password("passnoemi")
+							.build();
 
 			Kingdom kingdom = Kingdom.builder()
-              .name("My new Kingdom")
-              .build();
+							.name("Noemi's Kingdom")
+							.build();
```
Comment (kondfox) [inline src/main/java/com/greenfox/tribesoflagopus/backend/BackendApplication.java:34]: You shouldn't have such user creations here. If you need a test user somewhy in your database, add that manually there by SQL script.

File: src/main/java/com/greenfox/tribesoflagopus/backend/controller/KingdomController.java (hunk 1 above)
Comment (kondfox) [inline src/main/java/com/greenfox/tribesoflagopus/backend/controller/KingdomController.java:26]: Add `/` at the beginning of your path.

File: src/main/java/com/greenfox/tribesoflagopus/backend/controller/KingdomController.java (hunk 2 above)
Comment (kondfox) [inline src/main/java/com/greenfox/tribesoflagopus/backend/controller/KingdomController.java:31]: Add `/` at the beginning of your path.

File: src/main/java/com/greenfox/tribesoflagopus/backend/service/KingdomService.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  public ResponseEntity<JsonDto> modifyKingdom(
+      Long userId,
+      KingdomInputModifyDto kingdomInputModifyDto) {
+
+    if (!userRepository.exists(userId)) {
+      StatusResponse userNotFoundStatus = StatusResponse.builder()
+          .status("error")
+          .message("user_id not found")
+          .build();
+      return ResponseEntity.status(404).body(userNotFoundStatus);
```
Comment (kondfox) [inline src/main/java/com/greenfox/tribesoflagopus/backend/service/KingdomService.java:62]: Code duplication, you are doing the same a few rows above.

File: src/main/java/com/greenfox/tribesoflagopus/backend/service/LoginService.java [hunk 5]
```diff
@@ -4,68 +4,67 @@
...
+
+  private boolean inputUserNameExists() {
+    return userRepository.existsByUsername(inputUserName);
+  }
+
+  private boolean inputPasswordIsCorrect() {
+    return inputPassword.equals(userRepository.findByUsername(inputUserName).getPassword());
+  }
+
+  private UserDto createUserDto() {
```
Comment (kondfox) [inline src/main/java/com/greenfox/tribesoflagopus/backend/service/LoginService.java:61]: probably this should have a better place in your dto service.

File: src/main/java/com/greenfox/tribesoflagopus/backend/service/RegistrationService.java [hunk 6]
```diff
@@ -128,14 +106,13 @@ private Integer generateRandomNumber(int min, int max) {
     return randomNumber;
   }
 
-  private ResponseEntity<JsonDto> createUserDto(User user) {
+  private UserDto createUserDto() {
```
Comment (kondfox) [inline src/main/java/com/greenfox/tribesoflagopus/backend/service/RegistrationService.java:109]: This would be better on your dto service.

**Review by kondfox — APPROVED** [review]
OK, but pay attention to my comments.

**Pull request merged**